### PR TITLE
OCR

### PR DESCRIPTION
**Pull request by ishvik**
This is OCR for printed images with some modifications in images.

### CODE REVIEW

File: freshlybuiltimagebol/OCR_Printed_Text.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+from PIL import Image
+
+
+
+
+pic=cv2.imread("blurred1.jpg")
+
+
+
+
```
Comment (FreshlyBuilt) [inline freshlybuiltimagebol/OCR_Printed_Text.py:15]: remove extra space from the file.

File: freshlybuiltimagebol/OCR_Printed_Text.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+import numpy as np
+import matplotlib.pyplot as plt
+import matplotlib.image as mpimg
+import pytesseract
+from PIL import Image
+
+
+
+
+pic=cv2.imread("blurred1.jpg")
```
Comment (FreshlyBuilt) [inline freshlybuiltimagebol/OCR_Printed_Text.py:11]: Don't use name of the image or file directly.

File: freshlybuiltimagebol/OCR_Printed_Text.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    return image
+
+
+
+
+
+def to_text(image):
+    pytesseract.pytesseract.tesseract_cmd = r"C:\Program Files\Tesseract-OCR\tesseract.exe"
+    string_from_image = pytesseract.image_to_string(image,lang='eng')
+    print(string_from_image)
```
Comment (FreshlyBuilt) [inline freshlybuiltimagebol/OCR_Printed_Text.py:56]: Never use any type of output function while submitting a pull request to the main repository. As these output function can only be used during debugging phase only.

File: freshlybuiltimagebol/OCR_Printed_Text.py [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+
+
+
+pic=cv2.imread("blurred1.jpg")
+
+
+
+
+
```
Comment (FreshlyBuilt) [inline freshlybuiltimagebol/OCR_Printed_Text.py:16]: write down all of the functions under single class.

File: freshlybuiltimagebol/OCR_Printed_Text.py [hunk 5]
```diff
@@ -5,65 +5,37 @@
 import pytesseract
```
Comment (vishal2612200) [inline freshlybuiltimagebol/OCR_Printed_Text.py:5]: use from statement instead of import.

File: freshlybuiltimagebol/OCR_Printed_Text.py [hunk 6]
```diff
@@ -5,65 +5,37 @@
...
+        M = cv2.getRotationMatrix2D(center_of_pic, angle, 1.0)
+        image = cv2.warpAffine(image, M, (w, h),flags=cv2.INTER_CUBIC, borderMode=cv2.BORDER_REPLICATE)
+        return image
+    def shapness_blur(image):
+        sharpen_kernel = np.array([[-1,-1,-1], [-1,9,-1], [-1,-1,-1]])
+        image = cv2.filter2D(image, -1, sharpen_kernel)
+        return image
+    def to_text(image):
+        pytesseract.pytesseract.tesseract_cmd = r"C:\Program Files\Tesseract-OCR\tesseract.exe"
+        string_from_image = pytesseract.image_to_string(image,lang='eng')
```
Comment (vishal2612200) [inline freshlybuiltimagebol/OCR_Printed_Text.py:33]: add return string_from_image to the function.

File: freshlybuiltimagebol/OCR_Printed_Text.py [hunk 7]
```diff
@@ -5,65 +5,37 @@
...
-
-
-##plot image in output
-def plot_image(image):
-    plt.imshow(image)
-    plt.xticks([])
-    plt.yticks([])
+class image_to_text:
+    def remove_noise(image):
+        image = cv2.fastNlMeansDenoisingColored(image,None,20,10,7,21)
```
Comment (vishal2612200) [inline freshlybuiltimagebol/OCR_Printed_Text.py:10]: it will give you an error. cv2 not defined. use from statement for the import of cv2.

**Pull request merged**